samples/crypto/test_sample/README.rst

```diff
@@ -0,0 +1,258 @@
+|                    | HKDF SHA512 |                                                                            | `RFC 5869 - HMAC-based Extract-and-Expand Key Derivation Function (HKDF)`_ |
++--------------------+-------------+----------------------------------------------------------------------------+----------------------------------------------------------------------------+
+
+
+Requirements
+************
+
+* One of the following development boards:
+
+  * |nRF9160DK|
+  * |nRF5340DK|
+  * |nRF52840DK|
+  * |nRF52DK|
+  * |nRF51DK|
+
+* something else??
+
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/bluetooth/central_bas`
+
+.. include:: /includes/build_and_run.txt
+
+.. _crypto_test_testing:
+
+Testing
+=======
+
+1. Compile and program the application.
+#. Observe the result of the different test vectors in the log using an RTT viewer or a terminal emulator.
+   Make sure the test vector verdict is "passed"::
+
+      ***************************************
+      All Tests Done
+      X test vectors passed
+      0 test vectors failed
+      ***************************************
+#.
+
+Additional test cases and test vectors
+======================================
+
+Test cases and test vectors can be added to the test suite either by including additional source files or by extending the existing files.
+
+
+Test case
+---------
+
+A test case is a function designed to verify parts of the functionality of an nrf_crypto operation.
+Most nrf_crypto operations, like hash calculations and ECDH, have multiple test cases to be able to cover all features.
+A typical test case loops over all related test vectors, and logs the verdict of each.
+
+An example of a hash test case in a simplified form is shown below::
+
+   NRF_SECTION_DEF(test_vector_hash_data, test_vector_hash_t);
+   #define TEST_VECTOR_GET(i)      NRF_SECTION_ITEM_GET(test_vector_hash_data, test_vector_hash_t, (i))    /**< Get number of hash test vectors. */
+   #define TEST_VECTOR_COUNT       NRF_SECTION_ITEM_COUNT(test_vector_hash_data, test_vector_hash_t)       /**< Get test vector reference from array of test vectors. */
+
+   ret_code_t exec_test_case_hash(test_info_t * p_test_info)
+   {
+       uint32_t i;
+       ret_code_t err_code;
+       uint32_t hash_test_vector_count = TEST_VECTOR_COUNT;
+
+       for (i = 0; i < hash_test_vector_count; i++)
+       {
+           test_vector_hash_t * p_test_vector = TEST_VECTOR_GET(i);
+
+           // TODO: Add test code here.
+
+           // Will goto exit_test_vector if fails.
+           TEST_VECTOR_ASSERT(err_code);
+
+           p_test_info->tests_passed++;
+
+   exit_test_vector: ;
+       }
+       return NRF_SUCCESS;
+   }
+
+Registering a test case
+-----------------------
+
+A new test case must be registered to the test_case_data section using lib_section_vars.
```

Comment:
"using lib_section_vars" and the following line can be changed with this

using ITEM_REGISTER which registers this with the named section **test_case_hmac_data**.

Reply:
done